packages/@aws-cdk/aws-pipes-sources-alpha/lib/streamSource.ts

```diff
@@ -0,0 +1,145 @@
+import { SourceWithDlq } from '@aws-cdk/aws-pipes-alpha';
+import { Duration } from 'aws-cdk-lib';
+import { ITopic } from 'aws-cdk-lib/aws-sns';
+import { IQueue } from 'aws-cdk-lib/aws-sqs';
+import { OnPartialBatchItemFailure } from './enums';
+
+/**
+ * Base parameters for streaming sources.
+ */
+export interface StreamSourceParameters {
+  /**
+   * The maximum number of records to include in each batch.
+   *
+   * Minumum: 1
+   * Maxiumum: 10000
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-batchsize
+   * @default 1
+   */
+  readonly batchSize?: number;
+
+  /**
+    * Define the target queue to send dead-letter queue events to.
+    *
+    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-deadletterconfig
+    * @default - no dead-letter queue or topic
+    */
+  readonly deadLetterTarget?: IQueue | ITopic;
+
+  /**
+   * The maximum length of a time to wait for events.
+   *
+   * Minumum: Duration.seconds(0)
+   * Maxiumum: Duration.seconds(300)
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumbatchingwindowinseconds
+   * @default - the events will be handled immediately
+   */
+  readonly maximumBatchingWindow?: Duration;
+
+  /**
+   * (Streams only) Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
```

Comment:
The several times that I'm seeing `(Streams only)` makes me think we may have the wrong level of abstraction here. If this was scoped only to where it is relevant, this comment wouldn't be necessary.

Reply:
Ah, so doc weirdness rather than abstraction weirdness. I'm going to check in with the service team on this to make sure there's no actual weirdness that I'm missing here. Gimme a day or two to chat with them and I'll do the next round of reviews.

Reply:
I'm Jeff from the EventBridge Pipes service team. I'll have a chat with our doc writer about making this clearer. It's an internal implementation issue that probably shouldn't be exposed. The "streams pollers" are the ones that poll from DynamoDB and Kinesis. When it says "streams only" it's synonymous with "DynanmoDB and Kinesis sources only". 

As a concrete example, let's consider "MaximumRecordAgeInSeconds". That applies to DynamoDB and Kinesis sources (it's in the PipeSourceKinesisStreamParameters and PipeSourceDynamoDBStreamParameters data type) but not for SQS / Kafka / MQ. 

Do you have a suggestion on what I can change the docs to? I'm thinking of just killing "(streams only)".

Reply:
I agree with the killing that verbiage. But good to know they are grouped together, I think I implemented this _mostly_ correct (thanks you @nmussy). @redwheeler3 Once this PR is in, I'll knock out the other sources & targets.

Reply:
OK. I filed a ticket with our doc team to get that verbiage removed. It may take a while, but eventually it will be gone.

Reply:
@TheRealAmazonKendra I removed `(Streams only)` from the documentation.